### PR TITLE
ttl: add more metrics for TTL

### LINKED ISSUE #41154: Collect more metrics for TTL
We need to collect more inforamtion in the TTL panel, to allow the users to get full information for the TTL execution.

1. TiDB / TiKV CPU usage
2. TTL insert rows
3. TiKV disk usage.

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: YangKeao <yangkeao@chunibyo.icu>

### What problem does this PR solve?

Issue Number: close #41154

### What is changed and how it works?

1. Add a new metrics: `ttl_insert_rows`
2. Collect all needed information in the TTL panel. Though it's weird to show TiKV CPU/disk in the TiDB dashboard, it's more convenient for the users to get all information needed to debug the TTL.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

image

### CODE REVIEW

File: table/tables/tables.go [hunk 1]
```diff
@@ -1592,6 +1595,10 @@ func shouldWriteBinlog(ctx sessionctx.Context, tblInfo *model.TableInfo) bool {
 	return !ctx.GetSessionVars().InRestrictedSQL
 }
 
+func shouldIncreaseTTLCount(tblInfo *model.TableInfo) bool {
+	return tblInfo.TTLInfo != nil && tblInfo.TTLInfo.Enable
```
Comment (lcwangchao) [inline table/tables/tables.go:1599]: I think we only need to check `tblInfo.TTLInfo != nil` ?

File: table/tables/tables.go [hunk 2]
```diff
@@ -1592,6 +1595,10 @@ func shouldWriteBinlog(ctx sessionctx.Context, tblInfo *model.TableInfo) bool {
 	return !ctx.GetSessionVars().InRestrictedSQL
 }
 
+func shouldIncreaseTTLCount(tblInfo *model.TableInfo) bool {
```
Comment (lcwangchao) [inline table/tables/tables.go:1598]: Is it better to rename the method to `shouldIncreaseTTLMetricCount` ?

File: table/tables/tables.go (hunk 1 above)
Comment (YangKeao) [reply table/tables/tables.go:1599]: You're right.

**Review by lcwangchao — APPROVED** [review]

File: table/tables/tables.go (hunk 2 above)
Comment (YangKeao) [reply table/tables/tables.go]: Yes.

File: session/session_test.go [hunk 3]
```diff
@@ -89,3 +92,50 @@ func TestMetaTableRegion(t *testing.T) {
...
+	tk.MustExec("insert into t values (NOW())")
+	tk.MustExec("commit")
+	require.Equal(t, 4.0, MustReadCounter(t, metrics.TTLInsertRowsCount))
+
+	// rollback will remove all recorded TTL rows
+	tk.MustExec("begin")
+	tk.MustExec("insert into t values (NOW())")
+	tk.MustExec("insert into t values (NOW())")
+	tk.MustExec("rollback")
+	tk.MustExec("commit")
```
Comment (bb7133) [inline session/session_test.go:130]: This line can be removed.

Comment (YangKeao) [reply session/session_test.go]: Fixed.

**Review by lcwangchao — APPROVED** [review]

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**